sensor/kubernetes/eventpipeline/component/component.go

```diff
@@ -12,15 +12,30 @@ type PipelineComponent interface {
 	Stop(error)
 }
 
-// Resolver component that performs the dependency resolution
+// Resolver component that performs the dependency resolution. This is the core component of the pipeline, and the reason
+// why re-sync is no longer required in kubernetes informers. It receives inputs from any kubernetes handler (from
+// listeners in sensor/kubernetes/listener/resources) in the form of a ResourceEvent object. Its main responsibility
+// is to look at the DeploymentReference slice, and process any deployments that need to be updated. Deployment references
+// are based on resolver.DeploymentResolution callback functions, which are defined by the handlers.
+//
+// For example: given deployment-A which has a service account (sa-A), and RBAC resources attached to it. When any of the
+// RBAC resources bound to sa-A change (created, updated or deleted), deployment-A has to be reprocessed. This is required
+// because the storage.Deployment object contains properties that are based off RBAC resources. They also need reprocessing,
+// because the state of violations might change. Rather than reprocessing every deployment every minute, the Resolver component
+// will look at any DeploymentReference entries that handlers might have added, and enqueue those deployment for reprocessing.
```

Comment:
nit: pluralize the last "deployment" in the sentence

```suggestion
// will look at any DeploymentReference entries that handlers might have added, and enqueue those deployments for reprocessing.
```